### PR TITLE
[#8] Decide on the console output format

### LINKED ISSUE #8: Decide on the console output format
How to code warnings/errors.
What to include? (loc, filepath, module name, more?)

### PR DESCRIPTION
**Pull request by vrom911**
Resolves #8
image
image

### CODE REVIEW

File: src/Stan/Inspection.hs [hunk 1]
```diff
@@ -48,6 +55,20 @@ data Severity
...
 
+-- | Show 'Category' in a human-friendly format.
+prettyShowCategory :: Category -> Text
+prettyShowCategory cat = formatWith [magentaBg] $ "#" <> unCategory cat
+
+-- | Get the colour of the severity level.
+severityColour :: Severity -> Text
+severityColour = \case
+    Severe -> red
+    NotReallySevere -> yellow
```
Comment (chshersh) [inline src/Stan/Inspection.hs:66]: Looks like it's time to create an issue to think about possible severities and colours for them :slightly_smiling_face:

File: src/Stan/Inspection.hs [hunk 2]
```diff
@@ -48,6 +55,20 @@ data Severity
...
+
+-- | Show 'Severity' in a human-friendly format.
+prettyShowSeverity :: Severity -> Text
+prettyShowSeverity s = formatWith [severityColour s, bold] $ show s
```
Comment (chshersh) [inline src/Stan/Inspection.hs:70]: This is nice! :ok_hand:

File: src/Stan/Inspection.hs [hunk 3]
```diff
@@ -57,13 +78,20 @@ inspections =
...
             one $ Category "Partial"
         , inspectionSeverity = Severe
         }
     ]
+
+-- | Get the 'Inspection' by the given inspection 'Id'.
+getInspectionById :: Id Inspection -> Inspection
+getInspectionById insId = case find ((==) insId . inspectionId) inspections of
+    Just ins -> ins
+    -- TODO: how to handle unknown ids better?
```
Comment (chshersh) [inline src/Stan/Inspection.hs:96]: It sounds like a difficult problem... Let's create an issue and discuss it? Ideally, I'd like to have a mapping from `Id Inspection` to `Inspection` where we use known `inspectionId`. But I'm not sure we can convince the compiler to do this efficiently...

File: src/Stan/Inspection.hs [hunk 4]
```diff
@@ -57,13 +78,20 @@ inspections =
...
+            , "Use explicit pattern-matching over lists"
             ]
         , inspectionCategory =
             -- TODO: See issue #25: https://github.com/kowainik/stan/issues/25
             one $ Category "Partial"
         , inspectionSeverity = Severe
         }
     ]
+
+-- | Get the 'Inspection' by the given inspection 'Id'.
```
Comment (chshersh) [inline src/Stan/Inspection.hs:92]: Just to tell that the id should be known
```suggestion
-- | Get the 'Inspection' by the given known inspection 'Id'.
```

File: src/Stan/Observation.hs [hunk 5]
```diff
@@ -26,8 +34,68 @@ data Observation = Observation
...
+      where
+        n :: Int
+        n = srcSpanStartLine observationLoc
+
+        alignLine :: Int -> Text
+        alignLine x = Text.justifyRight 4 ' ' (show x) <> " ┃ "
+
+        getSourceLine :: Text
+        getSourceLine = decodeUtf8 $
+            BS.lines observationFileContent !! (n - 1)
```
Comment (chshersh) [inline src/Stan/Observation.hs:86]: I agree that the usage of partial functions here is okay. It's too much work to handle errors in here and report them, especially when they shouldn't happen..

File: src/Stan/Observation.hs [hunk 6]
```diff
@@ -26,8 +34,68 @@ data Observation = Observation
...
+
+        arrows :: Text
+        arrows = severityColour (inspectionSeverity inspection)
```
Comment (chshersh) [inline src/Stan/Observation.hs:89]: Awesome idea to use the same highlighting for arrows! :heart_eyes:

File: src/Stan/Analysis/Pretty.hs [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+import Relude.Extra.Map (toPairs)
+
+import Stan.Analysis (Analysis (..))
+import Stan.Observation (Observation (..), prettyShowObservation)
+
+import qualified Data.Text as Text
+
+
+{- | Shows analysed output of Stan work.
+This functions groups 'Observation's by 'FilePath' they are found in.
```
[the remark this answers is not in the document]
Comment (vrom911) [reply src/Stan/Analysis/Pretty.hs:24]: Let's discuss in here: https://github.com/kowainik/stan/issues/29

File: src/Stan/Inspection.hs (hunk 1 above)
Comment (vrom911) [reply src/Stan/Inspection.hs:66]: Good idea: https://github.com/kowainik/stan/issues/30

File: src/Stan/Inspection.hs (hunk 3 above)
Comment (vrom911) [reply src/Stan/Inspection.hs:96]: See https://github.com/kowainik/stan/issues/31

**Pull request merged**